moveit_ros/hybrid_planning/local_planner/include/moveit/local_planner/local_planner_component.h

```diff
@@ -46,43 +48,134 @@
 
 #include "trajectory_msgs/msg/joint_trajectory.hpp"
 
+#include <moveit/planning_scene_monitor/planning_scene_monitor.h>
+#include <moveit/robot_model_loader/robot_model_loader.h>
+#include <moveit/local_planner/constraint_solver_interface.h>
+#include <moveit/local_planner/trajectory_operator_interface.h>
+
+#include <tf2_ros/buffer.h>
+#include <tf2_ros/transform_listener.h>
+
 namespace moveit_hybrid_planning
 {
-// The possible hybrid planner states
-// TODO Use lifecycle node?
+// TODO Refactor and use repository wide solution
+template <typename T>
+void declareOrGetParam(const std::string& param_name, T& output_value, const T& default_value,
+                       const rclcpp::Node::SharedPtr& node)
+{
+  try
+  {
+    if (node->has_parameter(param_name))
+      node->get_parameter<T>(param_name, output_value);
+    else
+      output_value = node->declare_parameter<T>(param_name, default_value);
+  }
+  catch (const rclcpp::exceptions::InvalidParameterTypeException& e)
+  {
+    RCLCPP_ERROR_STREAM(node->get_logger(),
+                        "Error getting parameter \'" << param_name << "\', check parameter type in YAML file");
+    throw e;
+  }
+}
+
+/// Internal local planner states
+/// TODO Use lifecycle node?
 enum class LocalPlannerState : int8_t
```

Comment:
Are these generic enough to go into the message type?

Reply:
Probably, but I think this should be replaced by a lifecycle component node with integrated states in the long run. Do you think we should keep it this way in the mean time?

Reply:
No need to. Using this for node lifecycle management at some point makes sense to me.